<a href="https://colab.research.google.com/github/suuu004/Data_analysis/blob/week4/0402_week4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**과제 1**

시간대별 승객 수 리스트를 본인이 직접 5개 값(임의값)으로 바꿔 입력 후, 혼잡도 결과를 2회 실행 비교

In [ ]:
def congestion_rate(passengers, capacity):
  return (passengers / capacity) * 100

times = ["06시", "07시", "08시", "09시", "10시"]
passengers = [160, 100, 310, 130, 230]  # 임의값 변경 1
capacity = 160

i=0
while i < len(times):
  rate = congestion_rate(passengers[i], capacity)
  print(f"{times[i]} 혼잡도: {rate:.1f}%")
  i = i + 1

06시 혼잡도: 100.0%
07시 혼잡도: 62.5%
08시 혼잡도: 193.8%
09시 혼잡도: 81.2%
10시 혼잡도: 143.8%


In [ ]:
def congestion_rate(passengers, capacity):
  return (passengers / capacity) * 100

times = ["06시", "07시", "08시", "09시", "10시"]
passengers = [90, 219, 225, 180, 170]  # 임의값 변경 2
capacity = 160

i=0
while i < len(times):
  rate = congestion_rate(passengers[i], capacity)
  print(f"{times[i]} 혼잡도: {rate:.1f}%")
  i = i + 1

06시 혼잡도: 56.2%
07시 혼잡도: 136.9%
08시 혼잡도: 140.6%
09시 혼잡도: 112.5%
10시 혼잡도: 106.2%


**실행 결과**

입력값의 변화에 따라 각 시간대의 혼잡도가 실시간으로 계산되어 출력됨을 확인했고, 이를 통해 유동 인구 데이터에 따라 혼잡도 분석 결과가 유동적으로 변하는 것을 확인했다.

**과제 2**

혼잡도 분류 기준 변경 후 실행 결과 비교하기

In [ ]:
passengers = [50, 100, 150, 200, 300] # 승객 수 초기화

def level(rate):
  if rate >= 150:  # 변경 전
    return "매우 혼잡"
  elif rate >= 100: # 변경 전
    return "혼잡"
  elif rate >= 70:  # 변경 전
    return "보통"
  else:
    return "여유"

max_rate = -1
max_time = ""

i = 0
while i < len(times):
  rate = congestion_rate(passengers[i], capacity)
  if rate > max_rate:
    max_rate = rate
    max_time = times[i]
  print(times[i], level(rate))
  i = i + 1

print("가장 혼잡한 시간:", max_time, f"({max_rate: .1f}%)")

06시 여유
07시 여유
08시 보통
09시 혼잡
10시 매우 혼잡
가장 혼잡한 시간: 10시 ( 187.5%)


In [ ]:
def level(rate):
  if rate >= 180:  # 매우 혼잡 기준 변경(150% -> 180%)
    return "매우 혼잡"
  elif rate >= 130: # 혼잡 기준 변경(100% -> 130%)
    return "혼잡"
  elif rate >= 95:  # 보통 기준 변경(70% -> 95%)
    return "보통"
  else:
    return "여유"

max_rate = -1
max_time = ""

i = 0
while i < len(times):
  rate = congestion_rate(passengers[i], capacity)
  if rate > max_rate:
    max_rate = rate
    max_time = times[i]
  print(times[i], level(rate))
  i = i + 1

print("가장 혼잡한 시간:", max_time, f"({max_rate: .1f}%)")

06시 여유
07시 여유
08시 여유
09시 보통
10시 매우 혼잡
가장 혼잡한 시간: 10시 ( 187.5%)


**실행 결과**

혼잡도 등급 분류 기준(매우 혼잡 기준 150% → 180% 등)을 상향 조정한 결과, 동일한 승객 수임에도 불구하고 일부 시간대의 등급이 '보통'에서 '여유' 등으로 하향 평가되는 등 전체적인 혼잡도 판정이 완화되었다. 하지만 등급 기준 변경은 수치 계산식에 영향을 주지 않으므로, 가장 혼잡한 시간대(10시)와 실제 혼잡도(187.5%) 등의 분석 데이터 자체는 변함없이 유지됨을 확인했다.

**과제 3**

정원을 180, 200으로 변경 후 최고 혼잡 시간과 등급 실행 결과 비교하기

In [ ]:
def congestion_rate(passengers, capacity):
    return (passengers / capacity) * 100

def level(rate):
    if rate >= 150:
        return "매우 혼잡"
    elif rate >= 100:
        return "혼잡"
    elif rate >= 70:
        return "보통"
    else:
        return "여유"

times = ["06시", "07시", "08시", "09시", "10시"]
passengers = [120, 210, 260, 230, 150]
capacity = 180  # 과제 3: 정원 180으로 변경

max_rate = -1
max_time = ""

print(f"--- [정원 {capacity}명 기준 분석 결과] ---")
i = 0
while i < len(times):
    rate = congestion_rate(passengers[i], capacity)
    if rate > max_rate:
        max_rate = rate
        max_time = times[i]
    print(f"{times[i]} 혼잡도: {rate:.1f}% | 상태: {level(rate)}")
    i = i + 1

print("-" * 30)
print("가장 혼잡한 시간:", max_time, f"({max_rate:.1f}%)")

--- [정원 180명 기준 분석 결과] ---
06시 혼잡도: 66.7% | 상태: 여유
07시 혼잡도: 116.7% | 상태: 혼잡
08시 혼잡도: 144.4% | 상태: 혼잡
09시 혼잡도: 127.8% | 상태: 혼잡
10시 혼잡도: 83.3% | 상태: 보통
------------------------------
가장 혼잡한 시간: 08시 (144.4%)


In [ ]:
def congestion_rate(passengers, capacity):
    return (passengers / capacity) * 100

def level(rate):
    if rate >= 150:
        return "매우 혼잡"
    elif rate >= 100:
        return "혼잡"
    elif rate >= 70:
        return "보통"
    else:
        return "여유"

times = ["06시", "07시", "08시", "09시", "10시"]
passengers = [120, 210, 260, 230, 150]
capacity = 200  # 과제 3: 정원 200으로 변경

max_rate = -1
max_time = ""

print(f"--- [정원 {capacity}명 기준 분석 결과] ---")
i = 0
while i < len(times):
    rate = congestion_rate(passengers[i], capacity)
    if rate > max_rate:
        max_rate = rate
        max_time = times[i]
    print(f"{times[i]} 혼잡도: {rate:.1f}% | 상태: {level(rate)}")
    i = i + 1

print("-" * 30)
print("가장 혼잡한 시간:", max_time, f"({max_rate:.1f}%)")

--- [정원 200명 기준 분석 결과] ---
06시 혼잡도: 60.0% | 상태: 여유
07시 혼잡도: 105.0% | 상태: 혼잡
08시 혼잡도: 130.0% | 상태: 혼잡
09시 혼잡도: 115.0% | 상태: 혼잡
10시 혼잡도: 75.0% | 상태: 보통
------------------------------
가장 혼잡한 시간: 08시 (130.0%)


**실행 결과**

종합 예제 1의 정원(capacity)을 180명과 200명으로 상향 조정하여 분석한 결과, 분모인 정원이 커짐에 따라 동일 승객 대비 혼잡도가 감소하여 최고 혼잡 시간대의 등급이 '매우 혼잡'에서 '혼잡' 등으로 완화됨을 확인했다. 이는 공급 용량(정원)의 확대가 실질적인 혼잡도 등급 개선에 직접적인 영향을 미친다는 것을 보여준다.